ios/CodePush/CodePushPackage.m

```diff
@@ -391,16 +418,26 @@ + (NSDictionary *)getPackage:(NSString *)packageHash
     
     NSString *updateMetadataString = [NSString stringWithContentsOfFile:updateMetadataFilePath
                                                                encoding:NSUTF8StringEncoding
-                                                                  error:error];
+                                                                  error:&localError];
     
-    if (*error) {
+    if (!updateMetadataString) {
+        if (localError && error) {
+            *error = localError;
+        }
         return nil;
     }
     
     NSData *updateMetadata = [updateMetadataString dataUsingEncoding:NSUTF8StringEncoding];
-    return [NSJSONSerialization JSONObjectWithData:updateMetadata
+    id result = [NSJSONSerialization JSONObjectWithData:updateMetadata
                                            options:kNilOptions
-                                             error:error];
+                                             error:&localError];
+
+    if (!result) {
```

Comment:
Right.. what I meant was `NSJSONSerialization` takes care of not dereferencing null pointers for you since the documentation says the error param can be nullable. Ultimately, what I'm suggesting is just for brevity, since your code does achieve the same thing